### PR TITLE
[ShapePainter] Define TS types

### PR DESCRIPTION
**Pull request by D8H**
This is to prepare a next PR that will add an action to clear the shape painter at request instead of each frame.

### CODE REVIEW

File: Extensions/PrimitiveDrawing/shapepainterruntimeobject.ts [hunk 1]
```diff
@@ -19,12 +19,12 @@ namespace gdjs {
...
     /** The color (in RGB format) of the outline of the painted shape */
     outlineColor: RGBColor;
-    /** The opacity of the inner part of the painted shape */
-    fillOpacity: number;
-    /** The opacity of the outline of the painted shape */
-    outlineOpacity: number;
+    /** The opacity of the inner part of the painted shape, form 0 to 255 */
+    fillOpacity: integer;
+    /** The opacity of the outline of the painted shape, form 0 to 255 */
+    outlineOpacity: integer;
```
Comment (4ian) [inline Extensions/PrimitiveDrawing/shapepainterruntimeobject.ts:25]:
```suggestion
    /** The opacity of the inner part of the painted shape, from 0 to 255 */
    fillOpacity: float;
    /** The opacity of the outline of the painted shape, from 0 to 255 */
    outlineOpacity: float;
```

Typo + even if it's usually set to an integer, the opacity can (and will) be a float (when it's progressively updated for example)

File: Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts [hunk 2]
```diff
@@ -2,14 +2,15 @@ namespace gdjs {
...
-    constructor(runtimeObject, runtimeScene) {
+    constructor(
+      runtimeObject: gdjs.ShapePainterRuntimeObject,
+      runtimeScene: gdjs.RuntimeScene
+    ) {
       this._object = runtimeObject;
-      if (this._graphics === undefined) {
-        this._graphics = new PIXI.Graphics();
-      }
+      this._graphics = new PIXI.Graphics();
```
Comment (4ian) [inline Extensions/PrimitiveDrawing/shapepainterruntimeobject-pixi-renderer.ts:13]: 👍

**Review by 4ian — APPROVED** [review]
Great 👍

File: Extensions/PrimitiveDrawing/shapepainterruntimeobject.ts (hunk 1 above)
Comment (D8H) [reply Extensions/PrimitiveDrawing/shapepainterruntimeobject.ts]: As the implementation uses it more like an integer at this time, I find it less confusing.
Can we let the person that will change it to 0.0 to 1.0 make it float?

Comment (4ian) [reply Extensions/PrimitiveDrawing/shapepainterruntimeobject.ts]: Well the thing is that it's not an error to have a floating point value (like 254.5), so we should in theory have a float here?

**Comment by 4ian** [comment]
Thank you!

**Pull request merged**